### PR TITLE
feat(release): support distribution tags and custom release user info

### PR DESCRIPTION
**Pull request by varl**
- Add some basic docs (incremental improvements...)
- Move to GitHub Actions
- Allow releases from branches: `master`, `next`, `next-major`, `N.x`, `N.x.x`, `N.N.x`, `beta`, `alpha`
- Release from branch `next` publishes on the dist-tag `@next`, more info here: https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/distribution-channels.md
- Supports maintenance release patches on branches, e.g. `N.x` where `N` is a number. The `x` is literal, e.g. to release a patch on version 1 create the branch `1.x`, for `1.2` create `1.2.x`, etc.
- `beta` and `alpha` branches will be marked as `prerelease`.
- Allow `GIT_*` variables to be configured through the environment, previously they were hardcoded.
- Allow semantic release to read its own configuration file.

### CODE REVIEW

File: packages/app/src/commands/i18n/modernize.js [hunk 1]
```diff
@@ -11,8 +11,6 @@
  *   --log-missing-keys
  */
 const path = require('path')
-const fs = require('fs')
-const chalk = require('chalk')
```
Comment (varl) [inline packages/app/src/commands/i18n/modernize.js]: Lint rule: no unused

File: packages/app/src/helpers/modernize/createNewTranslationFiles.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ const createNewTranslationFiles = ({
     appendToExistingFiles,
 }) => {
     for (const language in translations) {
-        if (translations.hasOwnProperty(language)) {
+        if (Object.prototype.hasOwnProperty.call(translations, language)) {
```
Comment (varl) [inline packages/app/src/helpers/modernize/createNewTranslationFiles.js]: Lint rule: no access to prototype built-ins

File: packages/app/src/helpers/modernize/deleteLegacyFiles.js [hunk 3]
```diff
@@ -16,7 +17,7 @@ const deleteLegacyFiles = ({ translationFiles, languagesToTransform }) => {
             languagesToTransform.indexOf(language) !== -1
         ) {
             try {
-                const filePathToDelete = path.join(inDir, file)
+                const filePathToDelete = file
```
Comment (varl) [inline packages/app/src/helpers/modernize/deleteLegacyFiles.js]: Lint rule: no undefined variables

File: packages/cluster/src/commands/db_cmds/restore.js [hunk 4]
```diff
@@ -20,7 +20,7 @@ const run = async function(argv) {
         process.exit(1)
     }
 
-    const rest = await tryCatchAsync(
+    const res = await tryCatchAsync(
```
Comment (varl) [inline packages/cluster/src/commands/db_cmds/restore.js]: Lint rule: no unused variables

Also a bug where `res` is accessed later, but the variable is named `rest`.

File: packages/cluster/src/commands/list.js [hunk 5]
```diff
@@ -51,21 +51,17 @@ const run = async function(argv) {
     await Promise.all(
         clusters.map(async cluster => {
             const status = await getStatus(cluster)
-            cluster.status = formatStatus(status)
```
Comment (varl) [inline packages/cluster/src/commands/list.js]: Lint rule: possible race condition (mutating source while looping)

File: packages/cluster/src/helpers/db/restore.js [hunk 6]
```diff
@@ -30,7 +30,7 @@ const downloadDatabase = async ({ cache, dbVersion, update, url }) => {
     }
 }
 
-const restoreFromFile = async ({ cacheLocation, dbFile, dbVersion, name }) => {
```
Comment (varl) [inline packages/cluster/src/helpers/db/restore.js]: Lint rule: no unused variables

File: packages/create/src/builders/app.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 const { reporter, chalk } = require('@dhis2/cli-helpers-engine')
 
-const buildApp = async ({ name, ...argv }) => {
+const buildApp = async () => {
```
Comment (varl) [inline packages/create/src/builders/app.js]: Lint rule: no unused variables

File: packages/create/src/index.js [hunk 8]
```diff
@@ -12,7 +12,7 @@ const handler = async ({ type, name, ...argv }) => {
     }
     type = type.toLowerCase()
     switch (type) {
-        case 'cli':
+        case 'cli': {
```
Comment (varl) [inline packages/create/src/index.js]: Lint rule: lexical scope needed to use `const` in `case`

File: packages/create/src/index.js [hunk 9]
```diff
@@ -35,17 +35,6 @@ const handler = async ({ type, name, ...argv }) => {
 const command = {
     command: 'create <type> [name]',
     desc: 'Create various DHIS2 components from templates',
-    builder: yargs => {
```
Comment (varl) [inline packages/create/src/index.js]: Lint rule: no unused variables

File: packages/main/src/commands/debug/config.js [hunk 10]
```diff
@@ -1,4 +1,4 @@
-const run = ({ _, $0, raw, ...argv }) => {
+const run = ({ raw, ...argv }) => {
```
Comment (varl) [inline packages/main/src/commands/debug/config.js]: Lint rule: no unused variables

File: packages/utils/src/support/semantic-release-defer-release.js [hunk 11]
```diff
@@ -1,6 +1,4 @@
-const SemanticReleaseError = require('@semantic-release/error')
-
-const analyzeCommits = (config = {}, context) => {
+const analyzeCommits = (config, context) => {
```
Comment (varl) [inline packages/utils/src/support/semantic-release-defer-release.js]: Lint rule: no unused variables

File: packages/utils/src/support/semantic-release-update-deps.js [hunk 12]
```diff
@@ -21,17 +21,17 @@ const verifyConditions = (config = {}, context) => {
         throw new AggregateError(errors)
     }
 
-    validPackages.forEach(package => {
-        package.label = package.json.name || '<unnamed>'
```
Comment (varl) [inline packages/utils/src/support/semantic-release-update-deps.js]: Lint rule: package is a reserved keyword

File: packages/utils/src/support/semantic-release-update-deps.js [hunk 13]
```diff
@@ -21,17 +21,17 @@ const verifyConditions = (config = {}, context) => {
...
-            logger.log(`Package ${package.label} found at ${package.path}`)
+            logger.log(`Package ${pkg.label} found at ${pkg.path}`)
         }
     })
 
     context.packages = validPackages
 }
 
-const replaceDependencies = (pkg, listNames, packageNames, version) => {
+const replaceDependencies = ({ pkg, listNames, packageNames, version }) => {
```
Comment (varl) [inline packages/utils/src/support/semantic-release-update-deps.js]: Lint rule: max-params > 3

**Comment by varl** [comment]
Added comments to the lint rule resolutions.

**Comment by amcgee** [comment]
@varl this looks great!!  (and thanks for fixing all my lint bugs)

I'll take a deeper look at the code, but allowing `release` to work for non-master branches is clutch :-)

**Comment by amcgee** [comment]
We could merge this to `next` to test it, but I think this is fine :-D

**Comment by varl** [comment]
I want to see if it works, so `next` here we go.

**Pull request merged**